sklearn/ensemble/_hist_gradient_boosting/gradient_boosting.py

```diff
@@ -193,9 +204,92 @@ class weights.
         """
         return sample_weight
 
+    def _preprocess_X(self, X, *, reset):
+        """Preprocess and validate X.
+
+        Parameters
+        ----------
+        X : {array-like, pandas DataFrame} of shape (n_samples, n_features)
+            Input data.
+
+        reset : bool
+            Whether to reset the `n_features_in_` and `feature_names_in_ attributes.
+
+        Returns
+        -------
+        X : ndarray of shape (n_samples, n_features)
+            Validated input data.
+
+        known_categories : list of ndarray of shape (n_categories,)
+            List of known categories for each categorical feature.
+        """
+        # If there is a preprocessor, we let the preprocessor handle the validation.
+        # Otherwise, we validate the data ourselves.
+        check_X_kwargs = dict(dtype=[X_DTYPE], force_all_finite=False)
+        if not reset:
+            if self._preprocessor is None:
+                return self._validate_data(X, reset=False, **check_X_kwargs)
+            return self._preprocessor.transform(X)
+
+        self.is_categorical_, known_categories = self._check_categories(X)
```

Comment:
I could be worth to mention that whatever is under is executed during `fit`.

While it is already kind of know since `reset == True`, it could be a bit more explicit while reading.